samples/components/dialects/SQLDialect.tdd

```diff
@@ -0,0 +1,273 @@
+        <remote-type name='default' value='BOOLEAN' />
+      </local-type>
+    </format-column-definition>
+    <!--
+      format-create-table
+      Piece-by-piece formula for creating a table.
+      Predicates can be used with tokens that only correspond
+      to a certain type of table.
+      Predicates:
+        GlobalTemp
+        LocalTemp
+        AnyTemp
+        NoTemp
+      String substitution tokens:
+        %n - table name
+        %f - formatted column list
+      -->
+    <format-create-table>
+      <formula>CREATE </formula>
+      <formula predicate='GlobalTemp'>GLOBAL </formula>
+      <formula predicate='LocalTemp'>LOCAL </formula>
+      <formula predicate='AnyTemp'>TEMPORARY </formula>
+      <formula>TABLE %n (</formula>
+      <formula>%f</formula>
+      <formula>)</formula>
+      <formula predicate='AnyTemp'> ON COMMIT PRESERVE ROWS</formula>
+    </format-create-table>
+    <!--
+      format-drop-table
+      Format for dropping a table. %1 is the table name.
+      Each formula is executed as a separate statement.
+    -->
+    <format-drop-table>
+      <formula>DROP TABLE %1</formula>
+    </format-drop-table>
+    <!--
+      format-false
+      String literal or expression for boolean false
+    -->
+    <format-false value='(1=0)' />
+    <!--
+      format-if-then-else
+      Strategy for formatting if/then/else statement
+      [Default]Case: (CASE WHEN %1 THEN %2 ELSE %3 END)
+      IIF: Use IIF formula
+    -->
+    <format-if-then-else value='Case' />
+    <!--
+      format-index
+      Strategy for qualifying an index name in a CREATE INDEX statement
+      [Default]FullyQualified: CREATE INDEX "table"."index" ON "schema"."table" ("table"."column")
+      ColumnNameOnly: CREATE INDEX index ON "schema"."table" ("table"."column")
+    -->
+    <format-index value='FullyQualified' />
+    <!--
+      format-insert
+      Strategy for formatting an INSERT statement
+      [Default]Individual: Separate INSERT query for each row
+      Bulk: Single INSERT query can insert multiple rows at once
+    -->
+    <format-insert value='Individual' />
+    <!--
+      format-is-distinct
+      Defines a strategy for determining whether two values are distinct.
+      NoNullCheck: (lhs [!]= rhs)
+      Keyword: (lhs IS [NOT ]DISTINCT FROM rhs)
+      Operator: ([NOT (]lhs <=> rhs[)])
+      [Default]Formula: ((lhs [!]= rhs) OR[AND] (lhs IS [NOT] NULL AND[OR] rhs IS [NOT] NULL))
+    -->
+    <format-is-distinct value='Formula' />
+    <!--
+      format-null
+      Allows a type-specific expression for NULL. Tableau defaults to 'NULL'
+      where no type-specific formula is given.
+    -->
+    <format-null>
+      <local-type name='real' value='NULL' />
+      <local-type name='int' value='NULL' />
+      <local-type name='bool' value='NULL' />
+      <local-type name='date' value='NULL' />
+      <local-type name='datetime' value='NULL' />
+      <local-type name='localint' value='NULL' />
```

Comment:
local-type should be from the internal DataType enum (codegen/coredata/enums.data)

Reply:
Based on my manual testing, local-type is restricted within those values https://github.com/tableau/connector-plugin-sdk/blob/master/validation/tdd_latest.xsd#L102
If a type not in this list, tdd won't be loaded.

Reply:
It looks like in practice only the DataType enum values can ever be used.  See DataDrivenDialect::FormatNULL where Expression::ExprTypeFromDataType is used to go from DataType ->ExprType.  Maybe @erademacher can double check me here.

Reply:
Ross is right. Maybe I should've added a DataType enum to the XSD instead of letting ArgumentType cover both DataType and Expression::Type (which is basically a superset of DataType that includes local types for literals in expressions), but for now, it is what it is.

Local types don't really make sense in this context, so please remove them to avoid confusion. Also, it looks like you're missing 'str'.

Reply:
Removed local types in the latest commit, 'str' is also added.
Talked a bit with Evan, maybe we can create a follow-up story to add DataType enum to the XSD to let format-null use it, what is your opinion? @rosswbrown

Reply:
Yeah I think it makes sense to open something up to consider it at least.  It would be a breaking change at the XSD level, but in practice its already a runtime error, so I think there should be limited impact.  It would be 2020.4 or later timeframe either way.

Reply:
created a story as 1130681